circuits/lib/query/credentialAtomicQuerySig.circom

```diff
@@ -139,35 +147,46 @@ template CredentialAtomicQuerySig(IdOwnershipLevels, IssuerLevels, valueArraySiz
-    hashHi.inputs[3] <== issuerPubKeyY;
-    hashHi.out === issuerAuthHi;
-
-    // issuerClaim proof of existence (isProofExist)
+    component issuerSchemaCheck = verifyCredentialSchema();
+    for (var i=0; i<8; i++) { issuerSchemaCheck.claim[i] <== issuerAuthClaim[i]; }
+    issuerSchemaCheck.schema <== AUTH_SCHEMA_HASH;
+    // verify authClaim issued and not revoked
+    // calculate issuerAuthState
+    component issuerAuthStateComponent = getIdenState();
+    issuerAuthStateComponent.claimsTreeRoot <== issuerAuthClaimsTreeRoot;
+    issuerAuthStateComponent.revTreeRoot <== issuerAuthRevTreeRoot;
+    issuerAuthStateComponent.rootsTreeRoot <== issuerAuthRootsTreeRoot;
+
+    issuerAuthState <== issuerAuthStateComponent.idenState;
+
+
+    // issuerAuthClaim proof of existence (isProofExist)
     //
-    component smtIssuerAuthClaimExists = SMTVerifier(IssuerLevels);
-    smtIssuerAuthClaimExists.enabled <== 1;
-    smtIssuerAuthClaimExists.fnc <== 0; // Inclusion
-    smtIssuerAuthClaimExists.root <== issuerClaimsTreeRoot;
-    for (var i=0; i<IssuerLevels; i++) { smtIssuerAuthClaimExists.siblings[i] <== issuerAuthClaimMtp[i]; }
-    smtIssuerAuthClaimExists.oldKey <== 0;
-    smtIssuerAuthClaimExists.oldValue <== 0;
-    smtIssuerAuthClaimExists.isOld0 <== 0;
-    smtIssuerAuthClaimExists.key <== issuerAuthHi;
-    smtIssuerAuthClaimExists.value <== issuerAuthHv;
+    component smtIssuerAuthClaimExists = checkClaimExists(IssuerLevels);
+    for (var i=0; i<8; i++) { smtIssuerAuthClaimExists.claim[i] <== issuerAuthClaim[i]; }
+    for (var i=0; i<IssuerLevels; i++) { smtIssuerAuthClaimExists.claimMTP[i] <== issuerAuthClaimMtp[i]; }
+    smtIssuerAuthClaimExists.treeRoot <== issuerAuthClaimsTreeRoot;
+
+    // issuerAuthClaim proof of non-revocation
+    //
+    component verifyIssuerAuthClaimNotRevoked = checkClaimNotRevoked(IssuerLevels);
+    for (var i=0; i<8; i++) { verifyIssuerAuthClaimNotRevoked.claim[i] <== issuerAuthClaim[i]; }
+    for (var i=0; i<IssuerLevels; i++) {
+        verifyIssuerAuthClaimNotRevoked.claimNonRevMTP[i] <== issuerAuthClaimNonRevMtp[i];
+    }
+    verifyIssuerAuthClaimNotRevoked.noAux <== issuerAuthClaimNonRevMtpNoAux;
+    verifyIssuerAuthClaimNotRevoked.auxHi <== issuerAuthClaimNonRevMtpAuxHi;
+    verifyIssuerAuthClaimNotRevoked.auxHv <== issuerAuthClaimNonRevMtpAuxHv;
+    verifyIssuerAuthClaimNotRevoked.treeRoot <== issuerClaimNonRevRevTreeRoot;
 
     // issuerClaim  check signature
     component verifyClaimSig = verifyClaimSignature();
     for (var i=0; i<8; i++) { verifyClaimSig.claim[i] <== issuerClaim[i]; }
     verifyClaimSig.sigR8x <== issuerClaimSignatureR8x;
     verifyClaimSig.sigR8y <== issuerClaimSignatureR8y;
     verifyClaimSig.sigS <== issuerClaimSignatureS;
-    verifyClaimSig.pubKeyX <== issuerPubKeyX;
-    verifyClaimSig.pubKeyY <== issuerPubKeyY;
+    verifyClaimSig.pubKeyX <== issuerAuthClaim[2];//issuerPubKeyX;
+    verifyClaimSig.pubKeyY <== issuerAuthClaim[3];//issuerPubKeyY;
```

Comment:
There is something like `getPublicKeyFromClaim` template for this.

Reply:
updated